Source/Core/Core/IOS/SDIO/SDIOSlot0.cpp

```diff
@@ -27,6 +29,29 @@ SDIOSlot0Device::SDIOSlot0Device(Kernel& ios, const std::string& device_name)
 {
   if (!Config::Get(Config::MAIN_ALLOW_SD_WRITES))
     INFO_LOG_FMT(IOS_SD, "Writes to SD card disabled by user");
+
+  m_config_callback_id = Config::AddConfigChangedCallback([this] { RefreshConfig(); });
+  m_sd_card_inserted = Config::Get(Config::MAIN_WII_SD_CARD);
+}
+
+SDIOSlot0Device::~SDIOSlot0Device()
+{
+  Config::RemoveConfigChangedCallback(m_config_callback_id);
+}
+
+void SDIOSlot0Device::RefreshConfig()
+{
+  if (m_sd_card_inserted != Config::Get(Config::MAIN_WII_SD_CARD))
+  {
+    Core::RunAsCPUThread([this] {
+      const bool sd_card_inserted = Config::Get(Config::MAIN_WII_SD_CARD);
```

Comment:
Should this have a `Core::IsRunning()` check, just in case?

Reply:
My thinking was that the destructor of this class will deregister the callback when emulation ends. Or would that not work as intended?

Reply:
My thinking was that there might be some multithreading shenanigans where one thread was in the process of shutting down while another was calling RefreshConfig(), but thinking about it in that case we'd have worse problems anyway by writing to an already destructed instance. So... probably fine, then.